command/agent/alloc_endpoint.go

```diff
@@ -105,43 +106,70 @@ func (s *HTTPServer) ClientAllocRequest(resp http.ResponseWriter, req *http.Requ
 }
 
 func (s *HTTPServer) ClientGCRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
-	if s.agent.client == nil {
-		return nil, clientNotRunning
-	}
-
-	var secret string
-	s.parseToken(req, &secret)
-
-	// Check node write permissions
-	if aclObj, err := s.agent.Client().ResolveToken(secret); err != nil {
-		return nil, err
-	} else if aclObj != nil && !aclObj.AllowNodeWrite() {
-		return nil, structs.ErrPermissionDenied
+	// Get the requested Node ID
+	requestedNode := req.URL.Query().Get("node_id")
+
+	// Build the request and parse the ACL token
+	args := structs.NodeSpecificRequest{
+		NodeID: requestedNode,
+	}
+	s.parse(resp, req, &args.QueryOptions.Region, &args.QueryOptions)
+
+	// Determine the handler to use
+	useLocalClient, useClientRPC, useServerRPC := s.rpcHandlerForNode(requestedNode)
+
+	// Make the RPC
+	var reply structs.GenericResponse
+	var rpcErr error
+	if useLocalClient {
+		rpcErr = s.agent.Client().ClientRPC("Allocations.GarbageCollectAll", &args, &reply)
+	} else if useClientRPC {
+		rpcErr = s.agent.Client().RPC("ClientAllocations.GarbageCollectAll", &args, &reply)
+	} else if useServerRPC {
+		rpcErr = s.agent.Server().RPC("ClientAllocations.GarbageCollectAll", &args, &reply)
+	} else {
+		rpcErr = CodedError(400, "No local Node and node_id not provided")
+	}
+
+	if rpcErr != nil {
+		if structs.IsErrNoNodeConn(rpcErr) {
+			rpcErr = CodedError(404, rpcErr.Error())
+		}
 	}
 
-	s.agent.Client().CollectAllAllocs()
-	return nil, nil
+	return nil, rpcErr
 }
 
 func (s *HTTPServer) allocGC(allocID string, resp http.ResponseWriter, req *http.Request) (interface{}, error) {
-	var secret string
-	s.parseToken(req, &secret)
+	// Build the request and parse the ACL token
+	args := structs.AllocSpecificRequest{
+		AllocID: allocID,
+	}
+	s.parse(resp, req, &args.QueryOptions.Region, &args.QueryOptions)
 
-	var namespace string
-	parseNamespace(req, &namespace)
+	// Determine the handler to use
+	useLocalClient, useClientRPC, useServerRPC := s.rpcHandlerForAlloc(allocID)
 
-	// Check namespace submit-job permissions
-	if aclObj, err := s.agent.Client().ResolveToken(secret); err != nil {
```

Comment:
Is this ACL token checked for later/earlier?

Reply:
Yeah if you look at the RPC implementations you'll see it